### DISCUSSION TITLE
Using a single USB drive for two Gotek as drive 0 and 1

### DISCUSSION THREAD

**Discussion by BK560A** [keirf/flashfloppy#725] [Q&A]
Hi all,
I was wondering if it woud be possible to use a single USB drive when you use in a system two Gotek set as drive 0 and 1.If yes, what would be the correct circuit design?Thanks.

**Comment by keirf** [comment]
It is not supported by current firmware.

**Reply by BK560A** [reply]
Thanks for fast reply.Do you think this feature will be implemented someday (as well as the two drives emulation)?

**Reply by keirf** [reply]
Enhanced Gotek design makes allowance for it. https://github.com/keirf/FlashFloppy/wiki/Enhanced-Gotek

**Reply by BK560A** [reply]
Yes, I know enhanced Gotek can support second drive emulation when this feature wil be implemented in firmware.Anyway, for me it would be better to have two separate Gotek with one common USB drive.Thanks for your support.

**Reply by keirf** [reply]
Oh I see! There are no plans for that.

**Reply by BK560A** [reply]
OK, thanks for your support.

**Reply by n3pla2000** [reply]
I also am interested in having a single GOTEK act like two drives. Also one that can change data rates based on disk image size.

In a Tandy Color Computer drives are setup as a standard Shugart pins 10, 12, and 14 are drive selects 0, 1, and 2 while pin 16 is used as a motor select. In a PC, Pin 12 is still used as drive select  and pin 16 is used as a motor select on all drives, however due to the fact that pins 10 through 16  have a twist between the first and second drive connectors, pins 10 and 14 become the drive select and motor select of the second drive. On both systems pin 32 is the side or head select.

The Tandy Color Computer sees 35, 40 and 80 track drives pretty much the same. To use drive with higher tracks than 35 there is the need to change the DOS used by the Color Computer to a values that make the DOS use all tracks as normally it stops formatting, reads and writes at track 34. The first track is of course track 0, so 0-34 is 35 tracks. People back when this computer was still in production determined which values to poke into memory to allow both 40 and 80 track operations. A 720k drive achieved 360k of data storage when using a single side. Others figured out what values to change in DOS to write to the opposite side, however the DOS sees the opposite side as another drive. Using a program called 6 drive allowed the DOS to see 6 single sided drives, which in reality were 3 double sided drives. The odd numbered side being the opposing side of the even numbered drive.

Physical drive 0 became two drives, virtual drive 0, and with the side select active low it was also virtual drive 1. The same for physical drive 1 becoming virtual drives 2 and 3 as well as physical drive 2 becoming virtual drive 4 and 5. If the GOTEK could read pins 10, 12, and 14 to determine which physical drive, and pin 32 to determine which virtual drive you could implement dual drives one of two ways for the Tandy Color Computer.

The first way is for one disk image to contain both virtual drives. For example, the disk image would be accessed as tracks 0-34, 0-39 or 0-79 or the first half of the disk image who's length is determined according to it's track count. When the drive select is active low, the disk image would be accessed as tracks 35-69, 40-79, or 80-159 or the second half of the disk image. Another factor would be how many and sectors there are how many granules, and how many bytes per track. On 35 and 40 track drives all the sectors, granules, and bytes per track are basically the same Track 17 is the directory. On 80 track drives the directory is still on the same track 17, however depending on if the drive is high density (720k) or Quad density (1.2M or 1.44M) the sectors, granules and bytes will be same or different from the 35 or 40 track drives. Also when accessing a Quad density image the data rate between the GOTEK and the Tandy Color Computer needs to be doubled, some people have modified their controllers to handle the greater speed, however being able to allow or cancel the high speed data might be useful. Having emulated a Quad density drive in every way it would easier if it can also be accessed at the lower bit rate.

Second way is being able to "mount" two different disk images. The drive select determines which of two images are accessed. which are determined through the LCD and controls on the from the front panel. Doing this could be more versatile. One could have software of a 35 track image, yet have data on a 80 track image. Some original software was placed on 35 track drives (drive 0) and expected a second drive (drive 1) to have data, the software would not know the difference between 35, 40 or 80 track. only that is able access the data as many programs used calls to BASIC, EXTENDED BASIC and DOS ROM chips installed in the computer. Other software used internal routines and can never access 40 or 80 tracks without a rewrite anyway.

Either way, the size of the disk image would tell the GOTEK if the disk image is double/high or quad density. Double density and high density use the same data rate but because of the increased number of tracks allows for double the density other wise data rates and data per track is identical. Quad density however increases this. Again there are two ways to handle this, first increase data rates and granules, sectors, and or data bytes per track or second maintain the data rate, and increase granules, sectors, and or data bytes per track...

The second method of using two disk images to simulate the two sides on a diskette is in my opinion better, however the first might be easier to implement.
